### PR TITLE
Implement Retry-After Header

### LINKED ISSUE #433: Retry-After header needs implementation/argument
Some headers still don't have case classes. In my case I need Retry-After. It should take an argument. From the HTTP 1.1 spec:

http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html (subsection 37)

> The value of this field can be either an HTTP-date or an integer number of seconds (in decimal) after the time of the response.

Just a string is probably fine though.

### PR DESCRIPTION
**Pull request by cquiroz**
Fixes #433

### CODE REVIEW

File: core/src/main/scala/org/http4s/headers/Retry-After.scala [hunk 1]
```diff
@@ -1,5 +1,26 @@
...
+}
+
+/**
+  * Constructs a `Retry-After` header.
+  *
+  * The value of this field can be either an HTTP-date or an integer number of seconds (in decimal) after the time of the response.
+  *
+  * @param retry Either the date of expiration or seconds until expiration
+  */
+final case class `Retry-After`(retry: Either[Instant, Int]) extends Header.Parsed {
```
Comment (cquiroz) [inline core/src/main/scala/org/http4s/headers/Retry-After.scala]: This is probably worth discussing. `Retry-After` can be either a time or duration in seconds so the obvious type seems to be `Either[Instant, Int]` but I haven't seen any other header like that

Do you think this is a good approach?

Comment (rossabaker) [inline core/src/main/scala/org/http4s/headers/Retry-After.scala]: Yes.  Though I wonder whether `Int` should be `FiniteDuration`, as it represents a time.

File: core/src/main/scala/org/http4s/util/Renderable.scala [hunk 2]
```diff
@@ -37,6 +37,19 @@ object Renderer {
       writer << dateFormat.format(t)
 
   }
+
+  implicit val IntRenderer: Renderer[Int] = new Renderer[Int] {
```
Comment (rossabaker) [inline core/src/main/scala/org/http4s/util/Renderable.scala]: You can already render an `Int` directly.

**Comment by cquiroz** [comment]
Updated to use `FiniteDuration` rather than seconds

**Comment by cquiroz** [comment]
After PR #983 travis is happy and found a bug on scala 2.10. The last commit should fix it

**Pull request merged**